packages/celotool/src/lib/geth.ts

```diff
@@ -520,26 +575,67 @@ export const simulateClient = async (
     // randomly choose which gas currency to use
     const feeCurrencyGold = Boolean(Math.round(Math.random()))
 
-    let feeCurrency
-    if (!feeCurrencyGold) {
-      try {
-        feeCurrency = await kit.registry.addressFor(CeloContract.StableToken)
-      } catch (error) {
-        tracerLog({
-          tag: LOG_TAG_CONTRACT_ADDRESS_ERROR,
-          error: error.toString(),
-          ...baseLogMessage,
-        })
+    let feeCurrency, txOptions
+    try {
+      feeCurrency = feeCurrencyGold ? '' : await kit.registry.addressFor(CeloContract.StableToken)
+    } catch (error) {
+      tracerLog({
+        tag: LOG_TAG_CONTRACT_ADDRESS_ERROR,
+        error: error.toString(),
+        ...baseLogMessage,
+      })
+    }
+
+    try {
+      feeCurrency = feeCurrency || ''
+      baseLogMessage.feeCurrency = feeCurrency
+
+      const gasPriceMinimum = await kit.contracts.getGasPriceMinimum()
+
+      const gasPriceBase = feeCurrency
+        ? await gasPriceMinimum.getGasPriceMinimum(feeCurrency)
+        : await gasPriceMinimum.gasPriceMinimum()
+      let gasPrice = new BigNumber(gasPriceBase).times(2)
+
+      // Check if last tx was mined. If not, reuse the same nonce
+      if (lastTx === '' || lastNonce === 0) {
+        nonce = await kit.connection.nonce(kit.defaultAccount)
+      } else if ((await kit.connection.getTransactionReceipt(lastTx))?.blockNumber) {
+        nonce = await kit.connection.nonce(kit.defaultAccount)
+      } else {
+        nonce = (await kit.connection.nonce(kit.defaultAccount)) - 1
+        gasPrice = gasPrice.times(6)
+        console.warn(
+          `TX ${lastTx} was not mined. Replacing tx reusing nonce ${nonce} and gasPrice ${gasPrice}`
+        )
+      }
+
+      if (!feeCurrencyGold) {
+        txOptions = {
+          gasPrice: gasPrice.toString(),
+          feeCurrency,
+          nonce: nonce,
+        }
+      } else {
+        txOptions = {
+          gasPrice: gasPrice.toString(),
+          nonce: nonce,
+        }
```

Comment:
I believe this should work.  Alternatively, we could have feeCurrency be null rather than '' if using celo for the fee, and then we wouldn't need the `|| null`
```suggestion
      txOptions = {
        gasPrice: gasPrice.toString(),
        feeCurrency: feeCurrency || null,
        nonce: nonce,
      }
```

Reply:
Maybe I am missing something but refactoring in that way I need to replicate this `if` into `transferCeloGold` and `transferCeloDollars` (`kitGoldToken.transfer` and `kitStableToken.transfer` does not support `feeCurrency = null`).

Reply:
I see.  It looks like it's not the kit functions themselves that don't support it, but rather than `transferCeloGold` and `transferCeloDollars` in this same file.  Seems using `undefined` instead of `null` would work.  Alternatively, if for whatever reason we don't want to have `feeCurrency: undefined`, we could reduce the duplication with:

```
txOptions = {
    gasPrice: gasPrice.toString(),
    nonce,
}
if (feeCurrency) {
    txOptions.feeCurrency = feeCurrency
}	      
```